exporter/signalfxexporter/translation/converter.go

```diff
@@ -94,287 +61,288 @@ func NewMetricsConverter(logger *zap.Logger, t *MetricTranslator) *MetricsConver
+	out := make([]*sfxpb.DataPoint, 0, in.Len())
 
-	for k, v := range attribs {
-		dim := &sfxpb.Dimension{
-			Key:   k,
-			Value: v,
+	for i := 0; i < in.Len(); i++ {
+		inDp := in.At(i)
+		if inDp.IsNil() {
+			continue
 		}
-		dimensions = append(dimensions, dim)
+
+		dp := *basePoint
+		dp.Timestamp = timestampToSignalFx(inDp.Timestamp())
+		dp.Dimensions = labelsToDimensions(inDp.LabelsMap(), extraDims)
+
+		val := inDp.Value()
+		dp.Value.DoubleValue = &val
+
+		out = append(out, &dp)
 	}
-	return dimensions
+	return out
 }
 
-func fromOCMetricDescriptorToMetricType(ocType metricspb.MetricDescriptor_Type) *sfxpb.MetricType {
-	if ocType > metricspb.MetricDescriptor_UNSPECIFIED &&
-		ocType <= metricspb.MetricDescriptor_SUMMARY {
-		return ocDescriptorToMetricType[int(ocType)]
+func makeBaseDataPoint(m pdata.Metric) *sfxpb.DataPoint {
+	return &sfxpb.DataPoint{
+		Metric:     m.Name(),
+		MetricType: fromMetricDataTypeToMetricType(m),
 	}
-
-	return nil
 }
 
-func appendDistributionValues(
-	sfxDataPoints []*sfxpb.DataPoint,
-	sfxBaseDataPoint *sfxpb.DataPoint,
-	distributionValue *metricspb.DistributionValue,
-) []*sfxpb.DataPoint {
-
-	// Translating distribution values per symmetrical recommendations to Prometheus:
-	// https://docs.signalfx.com/en/latest/integrations/agent/monitors/prometheus-exporter.html#overview
-
-	// 1. The total count gets converted to a cumulative counter called
-	// <basename>_count.
-	// 2. The total sum gets converted to a cumulative counter called <basename>.
-	sfxDataPoints = appendTotalAndSum(
-		sfxDataPoints,
-		sfxBaseDataPoint,
-		&distributionValue.Count,
-		&distributionValue.Sum)
-
-	// 3. Each histogram bucket is converted to a cumulative counter called
-	// <basename>_bucket and will include a dimension called upper_bound that
-	// specifies the maximum value in that bucket. This metric specifies the
-	// number of events with a value that is less than or equal to the upper
-	// bound.
-	metricName := sfxBaseDataPoint.Metric + "_bucket"
-	explicitBuckets := distributionValue.BucketOptions.GetExplicit()
-	if explicitBuckets == nil {
-		return sfxDataPoints
-	}
-	bounds := explicitBuckets.Bounds
-	sfxBounds := make([]string, len(bounds)+1)
-	for i := 0; i < len(bounds); i++ {
-		sfxBounds[i] = float64ToDimValue(bounds[i])
-	}
-	sfxBounds[len(sfxBounds)-1] = infinityBoundSFxDimValue
+func fromMetricDataTypeToMetricType(metric pdata.Metric) *sfxpb.MetricType {
+	switch metric.DataType() {
+
+	case pdata.MetricDataTypeIntGauge:
+		return &sfxMetricTypeGauge
 
-	for i, bucket := range distributionValue.Buckets {
+	case pdata.MetricDataTypeDoubleGauge:
+		return &sfxMetricTypeGauge
 
-		// Adding the "upper_bound" dimension.
-		bucketDimensions := make([]*sfxpb.Dimension, len(sfxBaseDataPoint.Dimensions)+1)
-		copy(bucketDimensions, sfxBaseDataPoint.Dimensions)
+	case pdata.MetricDataTypeIntSum:
```

Comment:
Add:

```
if metric.IntSum().IsMonotonic() {
  return &sfxMetricTypeGauge
}
```

Reply:
Isn't it the opposite of this?  `IsMonotonic() == false` then `gauge`.

Reply:
Yes